### ISSUE TITLE
How to change master server

### ISSUE THREAD

**Issue by GYUMO** [moosefs/moosefs#267]
## So can any body give me some useable way or advice.
# Thanks！

**Comment by dragon2611** [comment]
I'm not sure if this is the right way to do it, it worked for me but at your own risk.etc

Shutdown the master cleanly, install moosefs-master on the new server copy /etc/mfs and /var/lib/mfs to the new server.

Make sure the master on the old server cannot start (Otherwise we'd have 2 servers who think they are the same master and will have different data as things change, that would be Very Bad!).  I actually went as far as to uninstall the moosefs-master package from the old machine.

Start the master on the new server.

Make sure the chunk-servers are pointing at the new server if the IP has changed.

**Comment by GYUMO** [comment]
You are so kind, and I think that can be useful to me.Thank you. If I success, I will share my experience.

**Comment by GYUMO** [comment]
> I'm not sure if this is the right way to do it, it worked for me but at your own risk.etc
> 
> optional: make a note of the number of blocks and metadata version .etc in the cgi interface
> 
> Shutdown the master cleanly, install moosefs-master on the new server copy /etc/mfs and /var/lib/mfs to the new server.
> 
> I'd shutdown or block the chunkservers from connecting at this point.
> 
> Make sure the master on the old server cannot start (Otherwise we'd have 2 servers who think they are the same master and will have different data as things change, that would be Very Bad!). I actually went as far as to uninstall the moosefs-master package from the old machine.
> 
> Start the master on the new server.
> 
> optional: Check the cgi interface on the new server should have the same number of blocks/metadata version - they will probably show as unreachable if you did shutdown/block the chunkservers.
> 
> Make sure the chunk-servers are pointing at the new server if the IP has changed.
> start the chunkservers if you did stop them.
> 
> Edit: Added some optional but recommended steps

You are so kind, and I think that can be useful to me.Thank you. If I success, I will share my experience.

**Comment by GYUMO** [comment]
> > I'm not sure if this is the right way to do it, it worked for me but at your own risk.etc
> > optional: make a note of the number of blocks and metadata version .etc in the cgi interface
> > Shutdown the master cleanly, install moosefs-master on the new server copy /etc/mfs and /var/lib/mfs to the new server.
> > I'd shutdown or block the chunkservers from connecting at this point.
> > Make sure the master on the old server cannot start (Otherwise we'd have 2 servers who think they are the same master and will have different data as things change, that would be Very Bad!). I actually went as far as to uninstall the moosefs-master package from the old machine.
> > Start the master on the new server.
> > optional: Check the cgi interface on the new server should have the same number of blocks/metadata version - they will probably show as unreachable if you did shutdown/block the chunkservers.
> > Make sure the chunk-servers are pointing at the new server if the IP has changed.
> > start the chunkservers if you did stop them.
> > Edit: Added some optional but recommended steps
> 
> You are so kind, and I think that can be useful to me.Thank you. If I success, I will share my experience.

should I close the metalogger server, and use mount to disconnect the client with the master node when I  close old master cleanly

**Comment by dragon2611** [comment]
Yes sorry I should have said that, I think when I did it the clients were off (They were VM's) but in my case that's because they were pretty useless without mooseFS anyway.

Otherwise best case they'll throw I/O errors when trying to access the MFS mount, worst case if files were open at the time it could corrupt data.

The metaloggers I didn't bother but then I had to reload them to change IP anyway, and it would have probably been safer to exit them cleanly before starting.

**Comment by GYUMO** [comment]
> em cleanly before starting.

Thank you for your answer.You are welcome!
